### ISSUE TITLE
Add IOptions support for AddElasticsearchClientInstrumentation

### ISSUE THREAD

**Issue by gao-artur** [open-telemetry/opentelemetry-dotnet-contrib#1013]
# Issue with OpenTelemetry.Instrumentation.ElasticsearchClient

**Is this a feature request or a bug?**

* [x] Feature Request
* [ ] Bug

Please add support for `IOptions` pattern. [Registration extension method guidance for library authors](https://github.com/open-telemetry/opentelemetry-dotnet/blob/main/docs/trace/extending-the-sdk/README.md#registration-extension-method-guidance-for-library-authors) defines

> * **DO** support the [.NET Options pattern](https://learn.microsoft.com/dotnet/core/extensions/options) and **DO** support [named options](https://learn.microsoft.com/dotnet/core/extensions/options#named-options-support-using-iconfigurenamedoptions). The Options pattern allows users to bind [configuration](https://learn.microsoft.com/dotnet/core/extensions/configuration) to options classes and provides extension points for working with instances as they are created. Multiple providers may exist in the same application for a single configuration and multiple components (for example exporters) may exist in the same provider. Named options help users target configuration to specific components.
> 
>   * Use the [Configure](https://learn.microsoft.com/dotnet/api/microsoft.extensions.dependencyinjection.optionsservicecollectionextensions.configure#microsoft-extensions-dependencyinjection-optionsservicecollectionextensions-configure-1(microsoft-extensions-dependencyinjection-iservicecollection-system-string-system-action((-0)))) extension to register configuration callbacks for a given name.
> 
>   * Use the [IOptionsMonitor<T>.Get](https://learn.microsoft.com/dotnet/api/microsoft.extensions.options.ioptionsmonitor-1.get) method to access options class instances by name.

**Comment by gao-artur** [comment]
While trying to implement this, I found that the method `TracerProviderBuilder.ConfigureServices` isn't available in `OpenTelemetry v1.2.0`, so I changed it to use `$(OpenTelemetryCoreLatestPrereleaseVersion)` (which is currently 1.4.0-rc4). But then it started to error on the [usage](https://github.com/open-telemetry/opentelemetry-dotnet-contrib/blob/main/src/OpenTelemetry.Contrib.Shared/Api/EnumerationHelper.cs#L93) of the `DynamicMethod` in the included `EnumerationHelper.cs`. It appears that [DynamicMethod](https://learn.microsoft.com/en-us/dotnet/api/system.reflection.emit.dynamicmethod?view=net-7.0) isn't available in `netstandard2.0`. But I'm wondering how it was working with version 1.2.0?
